### ISSUE TITLE
Adding support for a Fronius inverter

### ISSUE THREAD

**Issue by knackerbrot** [springfall2008/batpred#1325]
I am trying to run predbat/batpred in monitor mode so I can investigate adding support for my inverter/battery (Fronius Symo Gen24 with BYD HVM battery)

I've modified the apps.yaml but can't seem to get the addon to load. 

apps.yaml and relevent logs are attached. 
[apps-yaml.txt](https://github.com/user-attachments/files/16348177/apps-yaml.txt)
[predbad.log](https://github.com/user-attachments/files/16348178/predbad.log)

I've tried several different minor changes to the apps.yaml. I know I don't have a forecast from solcast set up yet, I'm just aiming to load the addon properly so I can investigate further. Apologies - I'm pretty clueless generally, 

Happy to provide more context if required - thank you

**Comment by gcoan** [comment]
you are missing a definition of inverter_type in the apps.yaml which tells Predbat what type of inverter you have.

Either start off with one of the existing inverter types that predbat knows about (that behaves similarly to your inverter), and customise from there, or define it as being a completely bespoke inverter type

see https://springfall2008.github.io/batpred/other-inverters/ - the last bit to define a custom inverter type which is what it looks like what you are doing?

**Comment by knackerbrot** [comment]
thanks! yep - it's my intention to define a custom inverter type. 

your advice was helpful: predbat runs and generates predbat_dashboard.yaml once I add `  inverter_type: "SF"` to apps.yaml.

I can now continue to investigate. 

Someone has already developed a tool that interacts with this inverter type via MODBUS that seems to be working: https://github.com/atbore-phx/sbam

The sbam tool has limited functionality so I'm hoping to transfer the MODBUS logic over from sbam to predbat.

Any further advice would be appreciated.

**Comment by springfall2008** [comment]
I think you need to split the work into two steps:

1. Get your inverter working in Home Assistant and confirm you can control it.
2. Configure Predbat to work with your inverter via the Home Assistant Controls

Its not clear to me if you have step 1 yet?

**Comment by knackerbrot** [comment]
Ok thanks. 

The inverter is being successfully controlled via homeassistant but it's currently through sbam. 

I will firstly focus on understanding exactly how sbam is managing to control the battery charging - appreciate the advice.

**Comment by georgbachmann** [comment]
@knackerbrot as I am currently thinking of getting a battery for the same inverter that you have. Just wanted to know what you are using now? sbam? Or batpred? And is everything working as expected?

**Comment by knackerbrot** [comment]
I'm not sure if there has been any progress on batpred for the Fronius inverter. sbam kind of works but is very basic and needed a major refactor (as of when I was looking at it which was about a year ago). Your system might be well suited to the limitations of sbam. Otherwise, your best bet would be to utilise sbam's competent integration with the Fronius inverter to build Fronius support into batpred. That's what I was hoping to do but lacked that patience and skill.

**Comment by georgbachmann** [comment]
Ok I see, thanks for your reply! So currently you don't have any smart battery-management? Cause I think it would shorten the return of investment quite a bit?!? I am still not sure if I should get the battery at all.
So to understand. Batpred does not currently support Fronius (I thought they are big? At least here in Austria I perceive them to be a large company... everybody I know has Fronius inverters). But the part of "how to communicate batpred <-> Fronius inverter" is what is still missing? But that part could in theory be taken from sbam and converted so that it works here as well?!? Did you dig in how much work that could be? I have no idea :-D

**Comment by knackerbrot** [comment]
Yes, as far as I understand the communication between batpred and Fronius is what's required. Theoretically, the fact that sbam manages to communicate with Fronius should mean that the solution 'just' needs to be transcribed to work with batpred. 

My own situation is currently that I'm busy with other things and the likely financial benefits of an effective weather/usage/etc. informed battery management system would be limited (for my particular case).

**Comment by chalfontchubby** [comment]
Thanks for the report. This issue has been open for over a year with no recent activity - we're volunteers working through a large backlog and need to filter what's still relevant. If this is still a problem for you, please let us know and we'll keep it open. Otherwise, we'll close it in a week (around 2026-08-28).

**Comment by chalfontchubby** [comment]
No further activity since our notice, so closing this as stale. If this is still a problem for you, feel free to reopen with a fresh reproduction case and we'll take another look.
